src/Components/Web/src/Sections/HeadSection.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.Sections
+{
+    /// <summary>
+    /// Renders content provided by <see cref="HeadContent"/> components.
+    /// </summary>
+    public sealed class HeadSection : ComponentBase
```

Comment:
I would have a faint preference for calling this `HeadOutput` rather than `HeadSection` because the term "section" seems a bit ambiguous - it's not obvious from the name whether this is the thing that provides content for the section, or the thing that emits the content that was provided elsewhere.

Reply:
That would also align with us having a future component called `SectionOutput`.